src/frontend/js/modules/authentication.js

```diff
@@ -8,15 +8,35 @@ angular.module('esn.authentication', ['esn.http'])
     const NETWORK_LAG = 15 * 1000;
     const CACHE_MAX_TTL = SERVER_TTL - NETWORK_LAG;
     const cachedToken = { promise: null, timestamp: 0 };
+    let cachedJWTPromise = null;
 
     return {
-      getNewToken
+      getNewToken,
+      getWebToken
     };
 
     function getNewToken(resetCache = false) {
       return resetCache ? getTokenNoCache() : getCachedToken();
     }
 
+    function getWebToken(reset = false) {
+      return reset ? requestNewWebToken() : getCachedWebToken();
+    }
+
+    function requestNewWebToken() {
+      cachedJWTPromise = esnRestangular.one('jwt/generate').post();
+
+      cachedJWTPromise.catch(() => {
+        cachedJWTPromise = null;
+      });
+
+      return cachedJWTPromise;
+    }
+
+    function getCachedWebToken() {
```

Comment:
This cache will last forever?

Reply:
@MichaelBailly @renaudboyer what do you think how long should the promise last for?

Reply:
i think it will last as long as the app is open

Reply:
How long does the JWT last?

Reply:
15 minutes